Lighting/Inc/lighting_controller.hpp

```diff
@@ -131,22 +192,58 @@ class LightingController {
 	 * @param colour : desired colour
 	 * @param brightness : desired brightness
 	 */
-	void set_domain_colour(ControlDomain domain, RGB_colour_t colour,
+	void set_domain_colour_and_brightness(ControlDomain domain, RGB_colour_t colour,
 			uint8_t brightness);
 
+	/**
+	 * This will set just the brightness of all the LEDs that are part of a control domain.
+	 *
+	 * @param domain : domain to be modified
+	 * @param colour : desired colour
+	*/
+	void set_domain_colour(ControlDomain domain, RGB_colour_t colour);
+
+	/**
+	 * This will set just the colour of all the LEDs that are part of a control domain.
+	 *
+	 * @param domain : domain to be modified
+	 * @param brightness : desired brightness
+	*/
+	void set_domain_brightness(ControlDomain domain, uint8_t brightness);
+
+	/**
+	 * This is a state machine that changes enabled leds, control domain colour
+	 * and brightness based on the state the drone is in.
+	 *
+	 * @param input : input that determines the next drone state and output.
+	 */
+	void transition_to(uint8_t input);
+
+
+	/*
+	 * Returns the state of the drone
+	 */
+	State get_drone_state();
+
 	// TODO: add function to "recolour domain" (enables & sets colour)
 	// Use this instead of set + enable?
 
 private:
-	static constexpr uint8_t NUM_LEDS = 6;
+	static constexpr uint8_t NUM_LEDS = 10;
+	State drone_state; //state of the drone.
```

Comment:
Re the comment above: See how `drone_state` can be confusing? We might be referring to the "drone state" currently, but "SEARCH" is not *strictly* a drone state, it's just a control state for our lighting boards?

`State` is also not a clear type here.